fesvr/htif.cc

```diff
@@ -82,8 +82,14 @@ htif_t::~htif_t()
 
 void htif_t::start()
 {
-  if (!targs.empty() && targs[0] != "none")
-      load_program();
+  if (!targs.empty() && targs[0] != "none"){
+      try{
+        load_program();
+      } catch( const std::runtime_error& err){
```

Comment:
It's not a good idea to use `std::runtime_error` and assume it could only come from the one new `throw` that you added. This could inadvertently catch unrelated things. The [C++ Core Guidelines](http://isocpp.github.io//CppCoreGuidelines/CppCoreGuidelines#Re-exception-types) best practice is to declare a new custom exception class.

It's a few extra lines of code but it will make this much safer.